check-pulp-msgs.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+#
+# Developer ..........: Waldirio M Pinheiro <waldirio@redhat.com>
+# Created at .........: 05/01/2021 
+# Purpose ............: Check the qpid messages coming from/to Pulp.
+# Tested Sat Version .: 6.6, 6.7, 6.8, 6.9
+#
+
+SRV_FQDN=$(hostname)
+CERT="/etc/pki/pulp/qpid/client.crt"
+LOG="/var/log/pulp_queue.log"
+
+qpid_stat()
+{
+  qpid-stat --ssl-certificate=$CERT -b amqps://localhost:5671 -q | grep -E "(resource_manager.*$SRV_FQDN.*|^  queu|^  =|resource_worker.*$SRV_FQDN.*|resource_manager )"
+}
+
+check_with_log()
+{
+while :
+do
+  date | tee -a $LOG
+  qpid_stat | tee -a $LOG
+  sleep 3
+done
+}
+
+check_with_no_log()
+{
+while :
+do
+  date
+  qpid_stat
+  sleep 3
+done
+}
```

Comment:
What I meant was:

```suggestion
write_log() {
  if [[ $LOG_OUTPUT == "Y" ]] ; then
    tee -a $LOG
  else
    cat
  end
}

check()
{
while :
do
  date | write_log
  qpid_stat | write_log
  sleep 3
done
}
```

Then you don't need the case at the end. Perhaps you can also solve it with
```bash
if [[ $LOG_OUTPUT == "Y" ]] ; then
  alias write_log="tee -a $LOG"
else
  alias write_log="cat"
fi
```

Reply:
I got your point, I'm using the case at the end to test the variable just once. On the way presented above, every 3 seconds I'll be using system resource to test if the LOG_OUTPUT == Y or N and we know this will not change.

Please, let me know what you think.

Thank you
Waldirio

Reply:
That's why I suggested the alias instead.